tasks/models.py

```diff
@@ -1,3 +1,39 @@
+from django.contrib.auth.models import User
 from django.db import models
 
-# Create your models here.
+
+class Tag(models.Model):
+    tag_name = models.CharField(max_length=50, unique=True, primary_key=True)
+
+    def __str__(self):
+        return self.tag_name
+
+
+class Task(models.Model):
+    task_code = models.CharField(max_length=8, unique=True, primary_key=True)
+    title = models.CharField(max_length=255)
+    author = models.ForeignKey(User, on_delete=models.CASCADE, to_field="username")
+    description = models.TextField()
+    tags = models.ManyToManyField(Tag)
+    score_mode = models.CharField(max_length=10)
```

Comment:
Is max_length required? Also maybe in the future we should define an enum and the "choices" parameter to limit the possible values. A possible solution is:
```
score_mode_enum = {
    SUM: "SUM",
    GROUP_MIN: "GROUP_MIN"
}

score_mode = models.CharField(choices = list(score_mode_enum.values()))
```

(however we should create a Enum class)

Reply:
In fact, I have done exactly what you are suggesting. I had a little confuse and I literally don't remember why I deleted it, hmm